Brokerages/Fxcm/FxcmBrokerage.cs

```diff
@@ -128,6 +135,92 @@ public override void Connect()
             // log in
             _gateway.login(loginProperties);
 
+            // create new thread to manage disconnections and reconnections
+            _connectionMonitorThread = new Thread(() =>
+            {
+                _lastReadyMessageTime = DateTime.UtcNow;
+
+                try
+                {
+                    while (!_cancellationTokenSource.IsCancellationRequested)
+                    {
+                        TimeSpan elapsed;
+                        lock (_lockerConnectionMonitor)
+                        {
+                            elapsed = DateTime.UtcNow - _lastReadyMessageTime;
+                        }
+
+                        if (!_connectionLost && elapsed > TimeSpan.FromSeconds(5))
+                        {
+                            _connectionLost = true;
+
+                            OnMessage(BrokerageMessageEvent.Disconnected("Connection with FXCM server lost. " +
+                                                                         "This could be because of internet connectivity issues. "));
+                        }
+                        else if (_connectionLost && elapsed <= TimeSpan.FromSeconds(5))
+                        {
+                            try
+                            {
+                                _gateway.relogin();
+                            }
+                            catch (Exception exception)
+                            {
+                                Log.Error(exception);
```

Comment:
If we throw an error during the call to `relogin` it looks like we'll swallow the exception and still set `_connectionLost=false` below. Maybe we could set `_connectionLost` to `false` inside the try after `_gateway.relogin()` and set it to `false` in the `catch` block